src/hotspot/share/opto/compile.cpp

```diff
@@ -2080,6 +2085,14 @@ void Compile::inline_incrementally_cleanup(PhaseIterGVN& igvn) {
     TracePhase tp(_t_incrInline_igvn);
     igvn.reset_from_gvn(initial_gvn());
     igvn.optimize();
+    // Reset failed generator in call node
```

Comment:
This is actually run in the incremental inline loop. Typically, we inline calls until we have too many nodes, at which time we do the cleanup with IGVN and IdealLoop and try again with the remaining calls in the queue. I think your approach is similar to we just retrying inlining a second time for all inlining failures.

As a result, I suggest you moving this to the main loop of `Compile::inline_incrementally`, on each iteration, we start inlining all calls in the `failed_once` queue, then continue with the main queue as normal. For `Compile::inline_incrementally_one`, each node that fails inlining is pushed into the `failed_once` queue (note that failed inline after `LiveNodeCountInliningCutoff` being broken should not be pushed into the `failed_once` queue and left in the original queue.